wagtail/admin/views/pages/bulk_actions/move.py

```diff
@@ -72,28 +71,43 @@ def prepare_action(self, pages):
             if not Page._slug_is_available(page.slug, destination, page=page):
                 pages_with_duplicate_slugs.append(page)
         if pages_without_destination_access or pages_with_duplicate_slugs:
+            # this will be picked up by the form
+            self.destination = destination
             return TemplateResponse(request, self.template_name, {
-                'pages_without_destination_access': pages_without_destination_access,
-                "pages_with_duplicate_slugs": pages_with_duplicate_slugs,
+                'pages_without_destination_access': [
+                    {'page': page, 'can_edit': page.permissions_for_user(self.request.user).can_edit()}
+                    for page in pages_without_destination_access
+                ],
+                "pages_with_duplicate_slugs": [
+                    {'page': page, 'can_edit': page.permissions_for_user(self.request.user).can_edit()}
+                    for page in pages_with_duplicate_slugs
+                ],
                 'destination': destination,
-                **self.get_context_data(destination=destination)
+                **self.get_context_data()
             })
 
-    def execute_action(cls, pages):
-        destination_page_id = cls.request.POST.dict().pop("chooser")
-        destination = get_object_or_404(Page, id=destination_page_id)
+    def get_execution_context(self):
+        return {
+            **super().get_execution_context(),
+            'destination': self.cleaned_form.cleaned_data['chooser'],
+        }
 
-        for page in pages:
-            if not page.permissions_for_user(cls.request.user).can_move_to(destination):
+    @classmethod
+    def execute_action(cls, objects, **kwargs):
+        destination = kwargs.get('destination', None)
+        if destination is None:
+            return
+        user = kwargs.get('user', None)
+        if user is None:
```

Comment:
note: we discussed that user is not mandatory in `execute_action` especially when used in a management command context. This and `destination` contradict that 🤔 (this applies to the other `execute_action`s
One to sort out with documentation in a separate PR

Reply:
I agree that requiring to input user in the management command is not ideal, but the requirement of user cannot be abstracted out from the action, because other than permission checks we also need the user for logging purposes in some cases. 
Maybe instead of making the default as `None` in `kwargs.get('user', None)`, we can default to any superuser in the database.

Reply:
Which actions is the user required for? As far as I know, all of the fundamental operations they run don't require a user.

I don't think defaulting to a random superuser is a good idea here - that will make for very confusing logs.

Reply:
@jacobtoppm see `execute_action`s in this PR (delete, move, publish, unpublish)

Reply:
Ah I assumed that for logging purposes, a user is must. Since that is not the case, in the cases where there is no 'user' in kwargs where there should have been, will it be safe to assume that the classmethod is being called from a management command, and treat all permission checks as true?

Reply:
@zerolab Yep, I'm referring more to the operations in core that they're calling, sorry for the confusion